x-pack/plugins/cases/public/components/cases_context/use_cases_features.tsx

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CasesContextValue } from '../../../common';
+import { useCasesContext } from './use_cases_context';
+
+interface UseCasesFeaturesReturn {
+  features: CasesContextValue['features'];
+  syncAlerts: boolean;
+}
+
+export const useCasesFeatures = (): UseCasesFeaturesReturn => {
+  const { features } = useCasesContext();
+  return { features, syncAlerts: features.alerts.sync };
```

Comment:
yes it is a good idea, the cases context will change the entire value sometimes, but this `features` property will never change the reference, so it is a good idea to memoize the return here 👍

Reply:
do you think it is a good idea to return both `features` and `syncAlerts`? I would prefer to use always the main `features` object, or always use named features such as this `syncAlerts`, but not both ways at the same time. Since we'll have more control over the way we get the features, and it will be easier to track all the places where a feature is used in the codebase.

Reply:
My idea behind exporting the `syncAlerts` is to avoid doing `const syncAlerts = features?.alerts?.sync` to every component. I exported the features in case in the future some component will need the whole features object. I am fine with returning only the named features!

Reply:
okay, yes we could return always a boolean here, or we could also ensure that all the features are always fully defined in the context (we would need to fix the _merge_ line commented below for that), 
But anyway I like exporting all named features as well, I would only change the naming a little bit to be more specific (`syncAlertsFeature` or `syncAlertsEnabled`) so we don't have to give them another name in the components.